pandas/tests/io/json/test_pandas.py

```diff
@@ -1606,3 +1607,13 @@ def test_emca_262_nan_inf_support(self):
             ["a", np.nan, "NaN", np.inf, "Infinity", -np.inf, "-Infinity"]
         )
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.filterwarnings("ignore:.*msgpack:FutureWarning")
+    def test_deprecate_numpy_argument_read_json(self):
+        # https://github.com/pandas-dev/pandas/issues/28512
+        expected = DataFrame([1, 2, 3])
+        with tm.assert_produces_warning(None):
```

Comment:
```suggestion
        with tm.assert_produces_warning(FutureWarning):
```

Should work and make things easier